src/reducers/states.ts

```diff
@@ -219,12 +219,29 @@ export enum Role {
  * Defines what chapters are available for usage.
  * For external libraries, see externalLibraries.ts
  */
-export const sourceChapters = [1, 2, 3, 4];
-export const sourceVariants = ['non-det'];
+export interface ISourceLanguage {
+  chapter: number;
+  variant: Variant;
+}
+
+export const sourceLanguages: ISourceLanguage[] = [
```

Comment:
For each language option, pair its chapter and variant. This is to keep the required data structures to a minimum and simplify logic in chapter selection, both via the menu and url.